### PR TITLE
Feature add books

### PR DESCRIPTION
**Pull request by jhkiehna**
Created the first model and controller for this API. And Some basic features tests to test them.

### CODE REVIEW

File: app/Infrastructure/AlgoliaSearchEngine.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Infrastructure;
+
+use Illuminate\Support\Collection;
+use Laravel\Scout\Engines\AlgoliaEngine;
+
+class AlgoliaSearchEngine extends AlgoliaEngine
```
Comment (joshforbes) [inline app/Infrastructure/AlgoliaSearchEngine.php]: You shouldn't have to do any of this advanced search engine stuff. There was a very specific reason that we had to write our own custom engine for construction jobs. But Laravel scout ships with an algolia driver built in. Everything should just work.

https://laravel.com/docs/5.6/scout

File: app/Models/Book.php [hunk 2]
```diff
@@ -3,13 +3,32 @@
...
     protected $fillable = [
         'title',
         'author'
     ];
+
+    protected $index;
+
+    public function __construct($params, Client $client)
+    {
+        $this->index = $client
```
Comment (joshforbes) [inline app/Models/Book.php]: Shouldn't have to do this - scout should just work

File: app/Providers/AppServiceProvider.php [hunk 3]
```diff
@@ -13,7 +17,14 @@ class AppServiceProvider extends ServiceProvider
      */
     public function boot()
     {
-        //
+        resolve(EngineManager::class)->extend('advancedAlgolia', function () {
```
Comment (joshforbes) [inline app/Providers/AppServiceProvider.php]: Shouldn't need this

File: app/Infrastructure/AlgoliaSearchEngine.php (hunk 1 above)
Comment (jhkiehna) [reply app/Infrastructure/AlgoliaSearchEngine.php]: I don't know why I got side tracked on this stuff in the first place. This should have been a separate PR

**Pull request merged**